src/js/App/RootApp.js

```diff
@@ -23,32 +24,46 @@ const isModule = (key, chrome) =>
   (key !== undefined && chrome?.activeSection?.group !== undefined && key === chrome?.activeSection?.group);
 
 const ShieldedRoot = memo(
-  ({ useLandingNav, hideNav, insightsContentRef, isGlobalFilterEnabled, initialized, remoteModule, appId }) => (
-    <Page
-      isManagedSidebar={!hideNav}
-      header={<PageHeader logoComponent="div" logo={<Header />} showNavToggle={!hideNav} headerTools={<HeaderTools />} />}
-      sidebar={hideNav ? undefined : <PageSidebar id="ins-c-sidebar" nav={useLandingNav ? <LandingNav /> : <SideNav />} />}
-    >
-      <div ref={insightsContentRef} className={isGlobalFilterEnabled ? '' : 'ins-m-full--height'}>
-        {isGlobalFilterEnabled && <GlobalFilter />}
-        {remoteModule && (
-          <main role="main" className={appId}>
-            {typeof remoteModule !== 'undefined' && initialized ? (
-              <ErrorBoundary>
-                {/* Slcaprum component does not react on config changes. Hack it with key to force new instance until that is enabled. */}
-                <ScalprumComponent fallback={<LoadingComponent />} LoadingComponent={LoadingComponent} key={remoteModule.appName} {...remoteModule} />
-              </ErrorBoundary>
-            ) : (
-              <Bullseye className="pf-u-p-xl">
-                <Spinner size="xl" />
-              </Bullseye>
-            )}
-          </main>
-        )}
-        <main className="pf-c-page__main" id="no-access"></main>
-      </div>
-    </Page>
-  ),
+  ({ useLandingNav, hideNav, insightsContentRef, isGlobalFilterEnabled, initialized, remoteModule, appId }) => {
+    const dispatch = useDispatch();
+    useEffect(() => {
+      const navToggleElement = document.querySelector('button#nav-toggle');
```

Comment:
Can we use `navRef` in here which would be passed to `PageHeader` to attach this `onClick` function?

Reply:
No we, unfortunately, can't. No refs are propagated. I've tried.